### PR TITLE
Adding a more comprehensive way of tracking queries across methods

### PR DESCRIPTION
**Pull request by Paultagoras**
So we kind of need to pair the query id and the offsets (otherwise it can be hard to trace them in the logs) - rather than go over the list of records again and again to parse it, I just pass through the "identifying" information we want to track.

I also cleaned up a few areas because changing them required a lot of copy + paste

### CODE REVIEW

File: src/main/java/com/clickhouse/kafka/connect/sink/processing/Processing.java [hunk 1]
```diff
@@ -43,9 +45,24 @@ public Processing(StateProvider stateProvider, DBWriter dbWriter, ErrorReporter
...
+     * @param topic
+     * @param partition
+     * @param minOffset
+     * @param maxOffset
      */
-    private void doInsert(List<Record> records) throws IOException, ExecutionException, InterruptedException {
-        dbWriter.doInsert(records, errorReporter);
+    private void doInsert(List<Record> records, String topic, int partition, long minOffset, long maxOffset) {
+        QueryIdentifier queryId = new QueryIdentifier(topic, partition, minOffset, maxOffset, UUID.randomUUID().toString());
+        if (records == null || records.isEmpty()) {
```
Comment (mzitnik) [inline src/main/java/com/clickhouse/kafka/connect/sink/processing/Processing.java:55]: Found it

**Review by mzitnik — CHANGES_REQUESTED** [review]

File: src/main/java/com/clickhouse/kafka/connect/sink/ClickHouseSinkConfig.java [hunk 2]
```diff
@@ -173,6 +173,7 @@ public ClickHouseSinkConfig(Map<String, String> props) {
         this.addClickHouseSetting("insert_quorum", "2", false);
         this.addClickHouseSetting("input_format_skip_unknown_fields", "1", false);
         this.addClickHouseSetting("wait_end_of_query", "1", false);
+        this.addClickHouseSetting("send_progress_in_http_headers", "1", false);
```
Comment (mshustov) [inline src/main/java/com/clickhouse/kafka/connect/sink/ClickHouseSinkConfig.java:176]: why do we need it?

**Review by mshustov — APPROVED** [review]

File: src/test/java/com/clickhouse/kafka/connect/sink/ClickHouseSinkTaskWithSchemaTest.java [hunk 3]
```diff
@@ -213,7 +213,11 @@ public void detectUnsupportedDataConversions() {
 
         ClickHouseSinkTask chst = new ClickHouseSinkTask();
         chst.start(props);
-        assertThrows(DataException.class, () -> chst.put(sr), "Did not detect wrong date conversion ");
+        try {
+            chst.put(sr);
+        } catch (RuntimeException e) {
+            assertTrue(Utils.getRootCause(e) instanceof DataException, "Did not detect wrong date conversion ");
```
Comment (mshustov) [inline src/test/java/com/clickhouse/kafka/connect/sink/ClickHouseSinkTaskWithSchemaTest.java:219]: Are these changes related? If not, let's not mix several changes in a single PR - it would greatly simplify the review.

**Review by mzitnik — APPROVED** [review]
